### PR TITLE
Quick fix for more informative old failure error message

### PR DESCRIPTION
**Pull request by sblausten**
https://trello.com/c/PvQxcRp3/308-improve-error-message-from-mailgun

We can use the reason string supplied in the drop webhook data to tell if this failed after retries (no mx or dns down) or whether it failed due to bounce, unsubscribe or spam complaint https://documentation.mailgun.com/en/latest/api-webhooks-deprecated.html#drops-parameters

### CODE REVIEW

File: src/main/scala/com/ovoenergy/event/gateway/manager/api/resources/MailgunResource.scala [hunk 1]
```diff
@@ -238,16 +238,23 @@ class MailgunResource(clock: Clock,
...
                         val rawData = EmailProgressedData(
                           timestamp,
                           recipient,
                           customParameters,
                           Some(messageId),
                           Dropped,
-                          Some(description.getOrElse("No failure reason supplied by Gateway"))
+                          Some(description.getOrElse {
+                            reason match {
+                              case Some("old") =>
```
Comment (SystemFw) [inline src/main/scala/com/ovoenergy/event/gateway/manager/api/resources/MailgunResource.scala]: where did you get these two strings? "old" and "hardfail" I mean

Comment (sblausten) [reply src/main/scala/com/ovoenergy/event/gateway/manager/api/resources/MailgunResource.scala]: They are specified in the docs - https://documentation.mailgun.com/en/latest/api-webhooks-deprecated.html#drops-parameters

Comment (sblausten) [reply src/main/scala/com/ovoenergy/event/gateway/manager/api/resources/MailgunResource.scala]: no idea if it will work or not though

Comment (SystemFw) [reply src/main/scala/com/ovoenergy/event/gateway/manager/api/resources/MailgunResource.scala]: worth a shot, worse case scenario is the same behaviour we have now

**Comment by ovo-comms-circleci** [comment]
This PR was successfully deployed to production by [CircleCI build 482](https://circleci.com/gh/ovotech/comms-gateway-event-manager/482) at 2020-01-07 11:17 GMT.

**Pull request merged**